File: erlang/raindrops/1/src/raindrops.erl
-module(raindrops).

-export([convert/1]).

convert(Number, Factor, Phrase, Acc) when (Number rem Factor =:= 0) ->
  [Phrase | Acc];

convert(Number, Factor, _, Acc) when (Number rem Factor =/= 0) ->
  Acc.

convert(Number) ->
  Drops = convert(Number, 3, "Pling",
          convert(Number, 5, "Plang",
          convert(Number, 7, "Plong", []))),

  if
    length(Drops) == 0 ->
      integer_to_list(Number);
    true ->
      lists:flatten(Drops)
  end.

File: ruby/raindrops/1/raindrops.rb

class Raindrops
  def self.convert(number)
    output = ''

    if number % 3 == 0
      output += 'Pling'
    end

    if number % 5 == 0
      output += 'Plang'
    end

    if number % 7 == 0
      output += 'Plong'
    end

    output = number.to_s if output.empty?

    output
  end
end
